sdk/aqueduct/utils.py

```diff
@@ -556,5 +557,12 @@ def generate_engine_config(config: Optional[FlowConfig]) -> EngineConfig:
                 integration_id=config.engine._metadata.id,
             ),
         )
+    elif isinstance(config.engine, LambdaIntegration):
+        return EngineConfig(
+            type=RuntimeType.LAMBDA,
+            lambda_config=LambdaEngineConfig(
```

Comment:
I wonder if we can have a EngineConfig(type=..., config=...) instead of a lambda_config variable, and infer what the config is for by looking at the engine type. Is that something we can do? Why do we choose to have a variable specifically for lambda_config instead?

Reply:
Also I don't see the EngineConfig file in this PR. Is this a change already in main?

Reply:
We need a way for the Golang DBWorkflowDag to identify both what type of engine it is using and the config for that engine integration, which is why we include the EngineConfig as part of the DBWorkflowDag.
```
type DBWorkflowDag struct {
	Id            uuid.UUID            `db:"id" json:"id"`
	WorkflowId    uuid.UUID            `db:"workflow_id" json:"workflow_id"`
	CreatedAt     time.Time            `db:"created_at" json:"created_at"`
	StorageConfig shared.StorageConfig `db:"storage_config" json:"storage_config"`
	EngineConfig  shared.EngineConfig  `db:"engine_config" json:"engine_config"`

	/* Field not stored in DB */
	Metadata  *workflow.Workflow                `json:"metadata"`
	Operators map[uuid.UUID]operator.DBOperator `json:"operators,omitempty"`
	Artifacts map[uuid.UUID]artifact.DBArtifact `json:"artifacts,omitempty"`
}
```

```
type EngineConfig struct {
	Type           EngineType      `yaml:"type" json:"type"`
	AqueductConfig *AqueductConfig `yaml:"aqueductConfig" json:"aqueduct_config,omitempty"`
	AirflowConfig  *AirflowConfig  `yaml:"airflowConfig" json:"airflow_config,omitempty"`
	K8sConfig      *K8sConfig      `yaml:"k8sConfig" json:"k8s_config,omitempty"`
	LambdaConfig   *LambdaConfig   `yaml:"lambdaConfig" json:"lambda_config,omitempty"`
}
```
The `EngineConfig` struct follows a similar pattern to how we create a field with different values. The Pydantic struct has to mirror this Golang struct.

The EngineConfig is declared in `sdk/aqueduct/config.py`. Lmk if this clears things up!